experimental/waf.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package experimental
+
+import (
+	"context"
+	"strings"
+
+	"github.com/corazawaf/coraza/v3/internal/corazawaf"
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+type Option func(*corazawaf.Options)
+
+// WithID sets the transaction ID
+func WithID(id string) Option {
+	return func(o *corazawaf.Options) {
+		o.ID = strings.TrimSpace(id)
+	}
+}
+
+// WithContext sets the transaction context, this is useful for passing
+// a context into the logger.
+// The transaction lifecycle isn't tied to the context lifecycle.
+func WithContext(ctx context.Context) Option {
+	return func(o *corazawaf.Options) {
+		o.Context = ctx
+	}
+}
+
+// WAFWithOptions is an interface that allows to create transactions
+// with options
+type WAFWithOptions interface {
```

Comment:
Also I wonder if it's worth extending the current interface so it adds the method to the existing rather than only having the new method (not sure it's possible but think so).

If it works perhaps this can just be called `experimental.WAF`, indicating the new API which we may make the default in a future major.

Either way we could also add a function to convert to the new interface so users can do it without worrying about type check. The function would just do an unchecked type cast.

Reply:
Not sure about it. I'd rather to expose experimental features over exposing a new experimental API (composed of many features). Unfortunately the way to expose experimental features is to create a new interface but that is much simpler than saying "the experimental WAF"